source/slang/slang-ir-insts-enum.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+
+namespace Slang
+{
+
+/* Bit usage of IROp is as follows
+
+           MainOp | Other
+Bit range: 0-10   | Remaining bits
+
+For doing range checks (for example for doing isa tests), the value is masked by kIROpMask_OpMask,
+such that the Other bits don't interfere. The other bits can be used for storage for anything that
+needs to identify as a different 'op' or 'type'. It is currently used currently for storing the
+TextureFlavor of a IRResourceTypeBase derived types for example.
+
+TODO: We should eliminate the use of the "other" bits so that the entire value/state
+of an instruction is manifest in its opcode, operands, and children.
+*/
+enum IROp : int32_t
+{
+
+#if 0 // FIDDLE TEMPLATE:
+% RAW(require("source/slang/slang-ir.h.lua").instEnums())
```

Comment:
The fact that `RAW(...)` is needed here is a little unfortunate. Ideally this would either be:

    % require(...).instEnums()

in the case where the function being called is itself inside a fiddle template and can thus generate code, or:

    % $(require(...).instEnums())

in the case where it is (for whatever reason) a function that returns a string.

Reply:
I'm also left wondering why things need to be factored into the `instEnums()` function, rather than just using a simple `% for` loop here.

Reply:
Unrelated to this PR: it would be good to set up the fiddle tool so that if a Lua function that generated code, like:

    % function stuff(a, b)
        class $a { $b };
    % end

gets called in a splice, like:

    namespace things { $(stuff(x,y)) }

then the string that gets spliced in is (or at least includes) whatever output the callee generated, and isn't just the return value of the call. Having this facility lets a user write their generation logic as a function including control flow operations, etc., while still having it be usable in templates as if it is just a function returning a string.

Reply:
> The fact that `RAW(...)` is needed here is a little unfortunate

ok, fixed